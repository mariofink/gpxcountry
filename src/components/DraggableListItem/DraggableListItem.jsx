let dragSource;

function handleDragStart(e, index) {
  e.nativeEvent.dataTransfer.effectAllowed = "move";
  // console.log("Drag Start", e, index);
  dragSource = index;
}

function handleDragOver(e) {
  if (e.nativeEvent.preventDefault) {
    e.nativeEvent.preventDefault(); // Allow dropping.
  }
}

function handleDragLeave(e) {
  // console.log("Drag Leave", e);
}

function handleDrop(e, index, callback) {
  console.log("DROP!", index, e);
  callback({ source: dragSource, target: index });
}

function handleDragEnd(e) {
  // console.log("Drag End", e);
}

const DraggableListItem = ({ children, index, onDrop }) => {
  return (
    <li
      className="leading-10 font-bold flex gap-2"
      draggable="true"
      onDragStart={(e) => handleDragStart(e, index)}
      onDragOver={handleDragOver}
      onDragLeave={handleDragLeave}
      onDrop={(e) => handleDrop(e, index, onDrop)}
      onDragEnd={handleDragEnd}
    >
      {index} {children}
    </li>
  );
};

export default DraggableListItem;
