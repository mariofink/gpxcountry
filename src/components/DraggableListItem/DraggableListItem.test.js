import React from "react";
import renderer from "react-test-renderer";
import DraggableListItem from "./DraggableListItem";

describe("DraggableListItem", () => {
  it("renders", () => {
    const tree = renderer
      .create(
        <ul>
          <DraggableListItem index="0" onDrop={() => {}}>
            My item
          </DraggableListItem>
        </ul>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
