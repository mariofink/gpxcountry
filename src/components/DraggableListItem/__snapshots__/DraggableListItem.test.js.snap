// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableListItem renders 1`] = `
<ul>
  <li
    className="leading-10 font-bold flex gap-2"
    draggable="true"
    onDragEnd={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
  >
    My item
  </li>
</ul>
`;
